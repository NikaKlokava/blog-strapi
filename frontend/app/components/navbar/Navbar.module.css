.navbar_container {
  display: flex;
  column-gap: 25px;
  justify-content: center;
  align-items: center;
}

.navbar_icon {
  display: none;
}

.navbar_title {
  font-size: 16px;
  letter-spacing: 5px;
}

.dot {
  width: 8px;
  height: 8px;
  border-radius: 3px;
  background-color: black;
  opacity: 0.7;
  text-decoration: none;
}

.link {
  transition: color 0.2s ease-in-out;
}

.link:hover {
  text-decoration: 3px underline dotted #E0B8C6;
  color: #BA8497;
}

@media screen and (max-width:1000px) {
  .navbar_container {
    column-gap: 15px;
  }

  .navbar_title {
    font-size: 12px;
  }

}

@media screen and (max-width:750px) {
  .navbar_container>* {
    display: none;
  }

  .navbar_icon {
    display: block;
    position: absolute;
    width: 40px;
    height: 40px;
    background-size: 40px 40px;
    background-color: aquamarine;
    align-self: flex-end;
  }
}