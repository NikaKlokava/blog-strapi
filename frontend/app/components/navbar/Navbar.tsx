"use client";

import { navbarItems } from "@/mocks/mocks";
import { useState } from "react";
import Link from "next/link";
import styles from "./Navbar.module.css";

export const Navbar = () => {
  const [active, setActive] = useState(false);
  return (
    <>
      <div className={active ? styles.navbar_active : styles.navbar_container}>
        <div className={styles.dot} />
        {navbarItems.map((item, index) => {
          return (
            <div key={item.path}>
              <Link href={item.path} key={item.path} className={styles.link}>
                <p className={styles.navbar_title}>{item.name}</p>
              </Link>
              <div className={styles.dot} key={index} />
            </div>
          );
        })}
        <div
          className={active ? styles.navbar_active_icon : styles.none}
          onClick={() => setActive(false)}
        />
      </div>
      <div
        className={!active ? styles.navbar_icon : styles.none}
        onClick={() => setActive(true)}
      />
    </>
  );
};
