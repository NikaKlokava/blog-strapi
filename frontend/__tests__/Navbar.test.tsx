import { Navbar } from "@/app/components/navbar/Navbar";
import { render } from "@testing-library/react";

describe("Test the Navbar component", () => {
  it("The Navbar component renders correctly", () => {
    const navbarSnap = render(<Navbar />);
    expect(navbarSnap).toMatchSnapshot();
  });
});
