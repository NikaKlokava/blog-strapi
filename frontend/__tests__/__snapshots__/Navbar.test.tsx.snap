// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Navbar component The Navbar component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar_container"
      >
        <div
          class="dot"
        />
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              HOME
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/sections/travel"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              TRAVEL
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/sections/life"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              LIFE
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/sections/styles"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              STYLES
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/the-best"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              THE BEST
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/insta-aesthetics"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              INSTA AESTHETICS
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
        <div
          class="navbar_item"
        >
          <a
            class="link"
            href="/about"
            rel="preload"
          >
            <p
              class="navbar_title"
            >
              ABOUT
            </p>
          </a>
          <div
            class="dot"
          />
        </div>
      </div>
      <img
        alt="burger_icon"
        class="navbar_icon"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar_container"
    >
      <div
        class="dot"
      />
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            HOME
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/sections/travel"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            TRAVEL
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/sections/life"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            LIFE
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/sections/styles"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            STYLES
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/the-best"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            THE BEST
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/insta-aesthetics"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            INSTA AESTHETICS
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
      <div
        class="navbar_item"
      >
        <a
          class="link"
          href="/about"
          rel="preload"
        >
          <p
            class="navbar_title"
          >
            ABOUT
          </p>
        </a>
        <div
          class="dot"
        />
      </div>
    </div>
    <img
      alt="burger_icon"
      class="navbar_icon"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
